optimize/client/src/modules/components/Select/Search.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import {ChangeEvent, FocusEvent, forwardRef, useContext, useEffect, useRef, useState} from 'react';
+import {Search as CarbonSearch} from '@carbon/react';
+import {MenuContext} from '@carbon/react/es/components/Menu/MenuContext';
+import {useMergedRefs} from '@carbon/react/lib/internal/useMergedRefs';
+
+import {t} from 'translation';
+
+import {useSelectContext} from './Select';
+
+import './Search.scss';
+
+interface SearchProps {
+  labelText?: string;
+  placeholder?: string;
+}
+
+export const Search = forwardRef<HTMLLIElement, SearchProps>(function MenuItemSearch(
+  {labelText = t('common.select').toString(), placeholder},
+  forwardRef
+) {
+  const context = useContext(MenuContext);
+  const menuItemRef = useRef<HTMLLIElement>(null);
+  const searchRef = useRef<HTMLInputElement>(null);
+  const ref = useMergedRefs<HTMLLIElement>([forwardRef, menuItemRef]);
+  const {children, setFilteredChildren} = useSelectContext();
+  const [search, setSearch] = useState('');
+
+  useEffect(() => {
+    context.dispatch({
+      type: 'registerItem',
+      payload: {
+        ref: menuItemRef,
+        disabled: false,
+      },
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I wasn't able make any custom components appear in carbon menu without registering it with the `context.dispatch`. I remember I tried simply rendering anything else than the menu items coming from carbon itself, but they were not showing up, because internally everything what is not registered is filtered out during render.
Feel free to try it out tho, maybe I am missing something here